org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/LaunchingPlugin.java

```diff
@@ -595,15 +603,36 @@ public void saving(ISaveContext context1) throws CoreException {
 			@Override
 			public void removedService(ServiceReference<IWorkspace> reference, IWorkspace service) {
 				service.removeSaveParticipant(ID_PLUGIN);
+				service.removeResourceChangeListener(LaunchingPlugin.this);
 				context.ungetService(reference);
 			}
 
 		};
 		fWorkspaceServiceTracker.open();
 
-		InstanceScope.INSTANCE.getNode(ID_PLUGIN).addPreferenceChangeListener(this);
+		Filter instanceLocationFilter = context.createFilter("(&(objectClass=" + Location.class.getName() + ")(" + Location.SERVICE_PROPERTY_TYPE //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
This is unfortunately not the right condition: https://github.com/eclipse-equinox/equinox/issues/932. Unclear how to defer until the workspace is set (i.e. the instance config is being set).